lib/ex_docker_build/dockerfile_parser.ex

```diff
@@ -3,10 +3,17 @@ defmodule ExDockerBuild.DockerfileParser do
   @continuation ~r/^.*\\\s*$/
   @instruction ~r/^\s*(\w+)\s+(.*)$/
 
-  @spec parse!(Path.t()) :: list(String.t()) | no_return()
-  def parse!(path) do
+  @spec parse!(Path.t() | String.t()) :: list(String.t()) | no_return()
+  def parse!(path_or_content) do
+    content =
+      if File.exists?(path_or_content) do
+        File.read!(path_or_content)
+      else
+        path_or_content
+      end
```

Comment:
yes, agree on the `parse_file!/1` and `parse_content!/1`.

> wonder if this could be used to somehow screw with the system

yeah, that's true. but how to prevent that? isn't that the job of the application using this lib to sanitize the input path? because we can't assume where the path must be

Reply:
A way the users of the lib can ensure a proper path is used, is to use `Path.join([VALID_PATH, USER_INPUT])` and not to expand the path, that way is limited to `VALID_PATH` and using things like `..` or `~/` won't work.